### PR TITLE
React: Merge "Using NodeJS For Your Backend" lesson with "Conclusion" 

### LINKED ISSUE #25998: <React, using Nodejs>: <out of place recommendations>
### Describe your suggestion

image

It says:

"For now, just keep in mind that this is something we’ll be discussing in depth later and head over to the next lesson and continue working through the curriculum."

but NodeJs *is* the next thing in the curriculum and not a distant thing in the future

I wonder if this lesson is out of place, or if it means to convey another message.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/react-new-using-nodejs-for-your-backend

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by OmarNaguib**
## Because

The "Using NodeJS For Your Backend" provided instructions that seemed out of place

## This PR

-edits the "Using NodeJS For Your Backend" in place (for ease of comparison)
-merges it with the conclusion lesson
-makes a react directory in the archive directory and moves the old lesson into it

## Issue

Closes #25998

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/conclusion/conclusion.md [hunk 1]
```diff
@@ -14,8 +14,19 @@ You're at a point in your React journey where you can begin learning about desig
...
 
+### Using NodeJS for your backend
+
+So, you've now learned a lot of juicy JavaScript, enabling you to make various components of your HTML/CSS page react to user actions and/or input. That's great! As you've seen, you can get your apps to do a lot of pretty cool things already, just using client-side JavaScript. You may have noticed, though, that there is still an important piece of the puzzle missing: unless you use [Local Storage](http://coding.smashingmagazine.com/2010/10/11/local-storage-and-how-to-use-it/), your app 'forgets' the user's preferences, as well as any other changes made, as soon as the page gets reloaded.
+
+While Local Storage is great, it's not ideal: it only stores data on the computer from which the user is accessing the page. This approach does not allow for the app to 'remember' anything when the same user tries to access it again from a different device. For that, you're going to need a real backend.
```
Comment (01zulfi) [inline react/conclusion/conclusion.md:21]: I feel this paragraph is redundant. Thoughts on removing it?

File: react/conclusion/conclusion.md [hunk 2]
```diff
@@ -14,8 +14,19 @@ You're at a point in your React journey where you can begin learning about desig
...
+
+### NodeJS to the rescue!
```
Comment (01zulfi) [inline react/conclusion/conclusion.md:23]: Not sure we need to keep this heading as well.. What do you think?

**Review by 01zulfi — CHANGES_REQUESTED** [review]
Left couple of comments below to make the changes concise.

Additionally, please make a copy of the current conclusion lesson into a separate file called `conclusion_ruby_on_rails.md`. Then, the changes you suggested in this PR should be reflected in a new file called `conclusion_full_stack_javascript.md`. This is because these changes should be shown in the Node path, not the Ruby on Rails path. The only way I can think of right now is to have separate files. in the curriculum repository.

Also putting this here: once this PR is merged, we'll need to update the website repository as well to reflect the changes.

**Comment by OmarNaguib** [comment]
> Additionally, please make a copy of the current conclusion lesson into a separate file called `conclusion_ruby_on_rails.md`. Then, the changes you suggested in this PR should be reflected in a new file called `conclusion_full_stack_javascript.md`. This is because these changes should be shown in the Node path, not the Ruby on Rails path. The only way I can think of right now is to have separate files. in the curriculum repository.
> 
done

File: react/conclusion/conclusion_full_stack_javascript.md.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+While Local Storage is great, it's not ideal: it only stores data on the computer from which the user is accessing the page. This approach does not allow for the app to 'remember' anything when the same user tries to access it again from a different device. For that, you're going to need a real backend.
+
+### NodeJS to the rescue!
+
+We are now ready to leave the frontend to learn all about how to store persistent data.
+NodeJS is a JavaScript runtime environment which will let you build an entire backend with JavaScript. We'll learn all about it next in our [NodeJS](https://www.theodinproject.com/paths/full-stack-javascript/courses/nodejs) course. With Node, we'll be able to add a bunch of cool features to our React apps like user authentication, data persistence, and more.
```
Comment (thatblindgeye) [inline react/conclusion/conclusion_full_stack_javascript.md.md:26]: I feel like these two sections can be combined into one + some other tweaks:

```suggestion
### Using a backend

As you've seen, you can get your apps to do a lot of pretty cool things already, just using client-side JavaScript. You may have noticed, though, that there is still an important piece of the puzzle missing: unless you use [Local Storage](http://coding.smashingmagazine.com/2010/10/11/local-storage-and-how-to-use-it/), your app 'forgets' the user's preferences, as well as any other changes made, as soon as the page gets reloaded.

While Local Storage is great, it's not ideal: it only stores data on the computer from which the user is accessing the page. This approach does not allow for the app to 'remember' anything when the same user tries to access it again from a different device. For that, you're going to need a real backend, which you'll learn all about next in our [NodeJS](https://www.theodinproject.com/paths/full-stack-javascript/courses/nodejs) course. With Node, we'll be able to add a bunch of cool features to your apps like user authentication, data persistence, and more.
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

**Review by 01zulfi — APPROVED** [review]
This looks good. Before we merge, we need to make relevant changes to the website repository as well. @OmarNaguib, are you willing to work on that? If not, that's okay as well.

**Comment by OmarNaguib** [comment]
> This looks good. Before we merge, we need to make relevant changes to the website repository as well. @OmarNaguib, are you willing to work on that? If not, that's okay as well.

Yeah I would like to @01zulfi

**Comment by 01zulfi** [comment]
@OmarNaguib Great, tag me when the website PR is ready for review. Do let me know if you need any help

**Pull request merged**